From 00b035d81b32e9321161529e40c15aa40f088201 Mon Sep 17 00:00:00 2001
From: Krzysztof Nowicki <krzysztof.a.nowicki@gmail.com>
Date: Fri, 27 Jan 2017 18:58:27 +0100
Subject: [PATCH 2186/3082] drm/amd/display: Fix MST physical ports always
 disconnected

Remove a false assumption that a cached EDID will be present whenever
the connector is in a connected state as this will only be true for
logical MST ports. For physical ports the EDID will never be cached,
which will cause them to always appear as disconnected.

This reverts commit 4ff8a88888de271bfb7750b2a5c68163848e2bf1
---
 drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm_mst_types.c | 8 --------
 1 file changed, 8 deletions(-)

diff --git a/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm_mst_types.c b/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm_mst_types.c
index deaae08..b263552 100644
--- a/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm_mst_types.c
+++ b/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm_mst_types.c
@@ -127,14 +127,6 @@ dm_dp_mst_detect(struct drm_connector *connector, bool force)
 			&master->mst_mgr,
 			aconnector->port);
 
-	/*
-	 * we do not want to make this connector connected until we have edid on
-	 * it
-	 */
-	if (status == connector_status_connected &&
-		!aconnector->port->cached_edid)
-		status = connector_status_disconnected;
-
 	return status;
 }
 
-- 
2.7.4

